cpp/src/copying/scatter.cu

```diff
@@ -47,7 +47,7 @@ __global__ void marking_bitmask_kernel(mutable_column_device_view destination,
                                        MapIterator scatter_map,
                                        size_type num_scatter_rows)
 {
-  size_type row = threadIdx.x + blockIdx.x * blockDim.x;
+  std::size_t row = threadIdx.x + blockIdx.x * blockDim.x;
 
   while (row < num_scatter_rows) {
```

Comment:
Comparison of signed and unsigned here too...

I think instead of just changing the types, we should replace this with a grid-stride for loop using a utility or an algorithm.  "no raw loops" -- this is exactly one of the reasons to avoid them.

I would advocate for at least adding a [grid-stride range helper like this](https://github.com/harrism/hemi#simple-grid-stride-loops) and using range-based for loops.

Reply:
Should we be bringing in Hemi (although quickly looking at it, I think it has the same issue) or should I roll something custom?   I'm somewhat uncomfortable implementing a change like this at this late stage.

Reply:
Should we be bringing in Hemi or should I roll something custom (possibly the range stuff copied from Hemi)?  I think it's a little spooky to be making changes to the overall loop structure of core kernels during burndown itself.